src/e3/collection/dag.py

```diff
@@ -546,19 +561,34 @@ def reverse_graph(self) -> DAG:
             result.update_vertex(node, data=self.vertex_data[node], enable_checks=False)
             for p in predecessors:
                 result.update_vertex(p, predecessors=[node], enable_checks=False)
-        try:
-            result.check()
-        except DAGError:
-            # Check detected
-            self.__has_cycle = True
-            raise
-        else:
-            # No cycle in the DAG
-            self.__has_cycle = False
+
+        if enable_checks:
```

Comment:
reuse the cache information before doing the checks. Otherwise it will be wrong for two reason: if the original graphs check was not done then you might end with __has_cycle set to None even if you do the check. If the original has has_cycle set to False you will run an unnecessary check.

Reply:
Done, thanks!